library/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt

```diff
@@ -83,6 +80,34 @@ public class ChannelsViewModelImpl(
         }.exhaustive
     }
 
+    private fun loadChannels() {
+        queryChannels().run {
+            channelsData = map(channels) { it.toState() }
+            loadingMoreData = map(loadingMore, ChannelsViewModel.State::LoadingNextPage)
+            endPageData = map(endOfChannels, ChannelsViewModel.State::EndPageReached)
+        }
+
+        stateMerger.addSource(channelsData, stateMerger::setValue)
+        stateMerger.addSource(loadingMoreData, stateMerger::setValue)
+    }
+
+    public fun hideChannel(channel: Channel) {
+        loadingData.postValue(ChannelsViewModel.State.Loading)
+        chatDomain.useCases.hideChannel(channel.cid, true).execute()
+        loadChannels()
```

Comment:
You don't need loadChannels. the channel list should update on it's own when the hide channel event is received.